instrumentation/mongo/mongo-3.1/testing/src/main/java/io/opentelemetry/instrumentation/mongo/v3_1/AbstractMongo31ClientTest.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.mongo.v3_1;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoTimeoutException;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.result.DeleteResult;
+import com.mongodb.client.result.UpdateResult;
+import io.opentelemetry.instrumentation.mongo.testing.AbstractMongoClientTest;
+import io.opentelemetry.instrumentation.test.utils.PortUtils;
+import java.util.ArrayList;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.Document;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public abstract class AbstractMongo31ClientTest
+    extends AbstractMongoClientTest<MongoCollection<Document>> {
+
+  protected abstract void configureMongoClientOptions(MongoClientOptions.Builder options);
+
+  private MongoClient client;
+
+  @BeforeAll
+  public void setupSpec() {
+    MongoClientOptions.Builder options =
+        MongoClientOptions.builder().description("some-description");
+    configureMongoClientOptions(options);
+    client = new MongoClient(new ServerAddress(host, port), options.build());
+  }
+
+  @AfterAll
+  public void cleanupSpec() {
```

Comment:
`cleanupSpec` is spock method names, I'd use `cleanup` to rename the AfterAll method, please modify other places together.